src/main/java/nextstep/courses/domain/SessionUsers.java

```diff
@@ -0,0 +1,43 @@
+package nextstep.courses.domain;
+
+import nextstep.courses.exception.CannotRegisterException;
+import nextstep.users.domain.NsUser;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class SessionUsers {
+    private final Set<NsUser> users;
+    private final Capacity capacity;
+
+    public SessionUsers(int capacity) {
+        this.capacity = new Capacity(capacity);
+        this.users = new HashSet<>();
+    }
+
+
+    public synchronized void updateCapacity(int value) {
+        if (value < users.size()) {
+            throw new IllegalStateException("변경하려는 수강 제한 인원이 현재 수강신청한 인원보다 작습니다");
+        }
+        capacity.update(value);
+    }
+
+    public synchronized void add(NsUser user) {
+        if (users.size() >= capacity.value()) {
```

Comment:
```suggestion
        if (capacity.isFull(users.size())) {
```

이와 같이 capacity에 메시지를 보내는 방식으로 구현하면 어떨까?